Source/SerializationTransformer.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  SerializationTransformer.swift
+//  Kakapo
+//
+//  Created by Alex Manzella on 27/06/16.
+//  Copyright © 2016 devlucky. All rights reserved.
+//
+
+import Foundation
+
+/// A closure that given a String as input output the transformed String, used to transform keys of the json
+public typealias KeyTransformer = (key: String) -> String
+
+/**
+ *  A protocol that special `Serializable` objects can adopt to transform other `Serializable` objects.
+ *  At the moment only key transforms are supported, see `SnakecaseTransformer` for a concrete implementations
+ */
+public protocol SerializationTransformer: CustomSerializable {
+    
+    /// The wrapped object type
+    associatedtype Wrapped: Serializable
+    
+    /// The wrapped `Serializable` object
+    var wrapped: Wrapped { get }
+    
+    /**
+     A function that given a String as input output the transformed String, used to transform keys of the json
+     
+     - parameter key: The key to transform
+     
+     - returns: Should return the transformed key, for example an `UppercaseTransformer` would return the uppercase key.
+     */
+    func transform(key key: String) -> String
+}
+
+extension SerializationTransformer {
+    
+    /**
+     `SerializationTransformer`by default serialize it's object using the given keyTransformer (if any) and it's own key transformer.
+     Its own keyTransformer must always be used before the one provided by the callers, so UppercaseTransformer(LowercaseTransformer(object)) would result in uppercase keys.
```

Comment:
And also the example `UppercaseTransformer(LowercaseTransformer(object))` does not provide any `KeyTransformer` right? It would be first a lowercase `transform` followed by an uppercase `transform` method.